posthog/models/filters/retention_filter.py

```diff
@@ -42,9 +44,23 @@ def __init__(self, data: Dict[str, Any] = {}, request: Optional[HttpRequest] = N
         elif not data:
             raise ValueError("You need to define either a data dict or a request")
         self.period = data.get(PERIOD, self.period)
-        self.target_entity = self._parse_target_entity(data.get(TARGET_ENTITY)) or self.target_entity
+        if len(self.entities) > 0:
+            self.target_entity = self.entities[0]
```

Comment:
Where does this condition apply now? Before entities array was used on returning entity at most